Source/WebKit/WebProcess/glib/WebProcessGLib.cpp

```diff
@@ -216,6 +217,9 @@ void WebProcess::platformInitializeWebProcess(WebProcessCreationParameters& para
         } else
             initializePlatformDisplayIfNeeded();
     }
+#else
+    initializePlatformDisplayIfNeeded();
+#endif
```

Comment:
Don't we want to keep the check for `!parameters.isServiceWorkerProcess` here? This is what I would have done:


```suggestion
    if (!parameters.isServiceWorkerProcess) {
#if USE(WPE_RENDERER)
        if (m_rendererBufferTransportMode.isEmpty()) {
            auto& implementationLibraryName = parameters.implementationLibraryName;
            if (!implementationLibraryName.isNull() && implementationLibraryName.data()[0] != '\0')
                wpe_loader_init(parameters.implementationLibraryName.data());
            PlatformDisplay::setSharedDisplay(PlatformDisplayLibWPE::create(parameters.hostClientFileDescriptor.release()));
        } else
#endif
            initializePlatformDisplayIfNeeded();
    }
```

Reply:
It would need more changes because isServiceWorkerProcess parameter is only defined for WPE_RENDERER. I wonder why this is only needed by WPE, I assumed this was added to not load any backend for service worker processes, that's why I left this specific to the old api.